lib/RobloxRenderer.spec.lua

```diff
@@ -73,6 +74,77 @@ return function()
 	end)
 
 	describe("updateHostNode", function()
+		it("should update node props and children", function()
+			local parent = Instance.new("Folder")
+			local key = "updateHostNodeTest"
+			local firstValue = "foo"
+			local newValue = "bar"
+
+			local _, defaultStringValue = getDefaultPropertyValue("StringValue", "Value")
+
+			local element = createElement("StringValue", {
+				Value = firstValue
+			}, {
+				ChildA = createElement("IntValue", {
+					Value = 1
+				}),
+				ChildB = createElement("BoolValue", {
+					Value = false,
+				}),
+				ChildC = createElement("StringValue", {
+					Value = "test",
+				}),
+				ChildD = createElement("StringValue", {
+					Value = "test",
+				})
+			})
+
+			local node = reconciler.createNode(element, parent, key)
+			RobloxRenderer.mountHostNode(reconciler, node)
+
+			-- Not testing mountHostNode's work here, only testing that the
+			-- node is properly updated.
```

Comment:
This brings up an interesting general testing philosophy. I like to assert that the construction was correct here anyways, since it's possible that a bug in mounting causes this test to pass mistakenly!

At the same time, there is an argument for only testing the thing you're actually trying to test. I'm not entirely sure!

Reply:
I wanted to cut down on clutter in this test - it's already pretty long, and verifying that mounting worked wouldn't provide any benefits on top of the existing `mountHostNode` tests.